sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -737,21 +733,22 @@ case class MapConcat(children: Seq[Expression]) extends ComplexTypeMergingExpres
   since = "2.4.0")
 case class MapFromEntries(child: Expression) extends UnaryExpression {
 
-  @transient
-  private lazy val dataTypeDetails: Option[(MapType, Boolean, Boolean)] = child.dataType match {
-    case ArrayType(
-      StructType(Array(
-        StructField(_, keyType, keyNullable, _),
-        StructField(_, valueType, valueNullable, _))),
-      containsNull) => Some((MapType(keyType, valueType, valueNullable), keyNullable, containsNull))
-    case _ => None
+  @transient private lazy val dataTypeDetails: Option[(MapType, Boolean, Boolean)] = {
+    child.dataType match {
+      case ArrayType(
+        StructType(Array(
+          StructField(_, kt, kn, _),
```

Comment:
nit: Is there any reason to change variable names? It would be good to minimize differences for review and ease of understanding.

Reply:
the motivation is described [here](https://github.com/apache/spark/pull/21352/files#r202983734). I will revert this piece of code shortly.